src/Sarif/SinglethreadedZipArchiveArtifactProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class SinglethreadedZipArchiveArtifactProvider : ArtifactProvider
+    {
+        public SinglethreadedZipArchiveArtifactProvider(ZipArchive zipArchive, IFileSystem fileSystem) : base(fileSystem)
+        {
+            var artifacts = new List<IEnumeratedArtifact>();
+
+            foreach (ZipArchiveEntry entry in zipArchive.Entries)
```

Comment:
Will this indirectly recurse?  Do we want it to?

(This might become obvious to me later, but tossing it out there in case it doesn't and I forget.)

Reply:
So, this should *not*. This is the zip archive's own enumerator, which handles traversing all zip contents. As far as I'm aware, the zip archive format does not allow features (such as a kind of symbolic link) that would result in any cycles/recursion.

Reply:
I was unclear and imprecise.  The zip archive doesn't know anything about the contents of its individual entries.  It's counterproductive to zip up a file that is already zipped, but you can.  And it probably happens in some cases where people are automating bulk archiving where other processes and/or user behavior may have caused some of the contents to be zipped up.

To get super concrete: if we're ever working through the entrys in "BulkArchive.zip", and find an entry in it such that entry.Name == "doublezipped.zip" --will we crack open doublezipped.zip?